src/core/thread/network_data_leader_ftd.cpp

```diff
@@ -317,53 +317,21 @@ template <> void Leader::HandleTmf<kUriCommissionerSet>(Coap::Message &aMessage,
 
 template <> void Leader::HandleTmf<kUriCommissionerGet>(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
 {
+    Error          error    = kErrorNone;
     Coap::Message *response = nullptr;
-    OffsetRange    offsetRange;
 
     VerifyOrExit(Get<Mle::Mle>().IsLeader() && !mWaitingForNetDataSync);
+    response = ProcessCommissionerGetRequest(aMessage);
 
-    response = Get<Tmf::Agent>().NewPriorityResponseMessage(aMessage);
     VerifyOrExit(response != nullptr);
```

Comment:
Should we set the `error` here from the `VerifyOrExit(response != nullptr);`? Since we now have added `LogWarnOnError(error, "send CommissionerGet response");` at `exit`.